src/Service/OrderStatusService.php

```diff
@@ -131,10 +131,10 @@ public function setOrderStatus($order, $statusId, $useExistingPayment = null, $t
 			$this->mailService->sendNewOrderMail($order, $statusId);
 		}
 
-		if (Configuration::get('MOLLIE_MAIL_WHEN_'.Tools::strtoupper($status))) {
-			$history->addWithemail(true, $templateVars);
-		} else {
+		if ('0' === Configuration::get('MOLLIE_MAIL_WHEN_'.Tools::strtoupper($status))) {
```

Comment:
`!(int) Configuration::get('MOLLIE_MAIL_WHEN_'.Tools::strtoupper($status))` can we use this ?

Reply:
if the value exists in configuration it returns 1 or 0, but if it doesnt it returns false. And i only need to check if its 0. So thats why no we can't :/ Tried to find another way but couldn't think of one without refactoring a lot of code.